src/Cli/dotnet/Installer/Windows/PipeStreamMessageDispatcherBase.cs

```diff
@@ -65,7 +65,7 @@ public void Connect()
                 if (_pipeStream is NamedPipeServerStream)
                 {
                     Task connectTask = Task.Factory.StartNew(() => ((NamedPipeServerStream)_pipeStream).WaitForConnection());
-                    connectTask.Wait(ConnectionTimeout);
+                    Task.Run(() => connectTask.Wait(ConnectionTimeout));
 
                     if (!connectTask.IsCompleted)
```

Comment:
This is a bug. The `Task.Run` will return immediately, before the `conntectTask.Wait` is even executed, so there's effectively zero chance that `connectTask` is complete. I didn't look at the test results, but I would't be surprised if this is a cause of a lot of the test failures.